examples/Microsoft.Spark.CSharp.Examples/Sql/Batch/Logging.cs

```diff
@@ -0,0 +1,142 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Microsoft.Spark.Sql;
+
+namespace Microsoft.Spark.Examples.Sql.Batch
+{
+    /// <summary>
+    /// An example demonstrating log processing.
+    /// Includes UDFs, regular expressions, and Spark SQL.
+    /// </summary>
+    internal sealed class Logging : IExample
+    {
+        /* Regular expression for log pattern matching later
+           Example Apache log line:   
+           64.242.88.10 - - [07/Mar/2004:16:47:12 -0800] "GET /robots.txt HTTP/1.1" 200 68
+           1:IP   2:client   3:user   4:date time   5:method   
+           6:req   7:proto   8:respcode   9:size */
+        static readonly string s_apacheRx =
+                "^(\\S+) (\\S+) (\\S+) \\[([\\w:/]+\\s[+\\-]\\d{4})\\] \"(\\S+) (\\S+) (\\S+)\" (\\d{3}) (\\d+)";
+
+        public void Run(string[] args)
+        {
+            if (args.Length != 1)
+            {
+                Console.Error.WriteLine(
+                    "Usage: Logging <path to Apache User Logs>");
+                Environment.Exit(1);
+            }
+
+            SparkSession spark = SparkSession
+                .Builder()
+                .AppName("Apache User Log Processing")
+                .GetOrCreate();
+
+            // Read input log file and display it
+            var df = spark.Read().Text(args[0]);
+            df.Show();
+
+            // Step 1: UDF to determine if each line is a valid log entry
+            // Remove any invalid entries before further filtering
+            spark.Udf().Register<string, string, bool>(
+                "GeneralReg",
+                (log, type) => Regex.IsMatch(log, s_apacheRx));
+
+            df.CreateOrReplaceTempView("Logs");
+
+            // Apply the UDF to get valid log entries
+            DataFrame generalDf = spark.Sql(
+                "SELECT logs.value, GeneralReg(logs.value, 'genfilter') FROM Logs");
+
+            generalDf.Show();
+
+            // Only keep log entries that matched the reg ex
+            generalDf = generalDf.Filter(generalDf["GeneralReg(value, genfilter)"] == true);
+            generalDf.Show();
```

Comment:
Looks like this goes through the table twice, is this the behavior you want ?  1st time will create the column genfilter, the second will filter.  Will going through the table one time with the filter work ?

Reply:
I had initially wanted it to go through twice, but now since the output is rather long, I think it'd be cleaner to just go through once with the filter. Going to make that adjustment.